### PR TITLE
SQL View Updates & Auto Apply

### LINKED ISSUE #5114: Grafana SQL View Updates/Fixes
## Background  
We’ve identified malformed records in the T3 and T7 file types that are causing downstream issues in both administrative and digital views. Specifically:

- In `tanf_t3`, `tribal_tanf_t3`, and `ssp_m3` (including admin views), records where `family_affiliation = 0` **and** `sex = 0` should be filtered out. These combinations indicate improperly formatted T3 records.
- In `tanf_t7`, `tribal_tanf_t7`, and `ssp_m7` (including admin views), records where `family_month = 0` **and** `tdrs_section_indicator` is **not equal to** 1 or 2 should be filtered out. These combinations signal improperly formatted T7 records.
- Additionally, there is a known issue where `age` indicator variables may error when `date_of_birth` is missing or invalid. This needs to be addressed in the view logic.

### Why It Matters  
Filtering out these invalid records helps ensure that downstream systems—such as reporting pipelines, admin dashboards, and validation tools—do not misrepresent or misinterpret corrupted data. This improves overall data quality, maintains trust in system outputs, and prevents errors that could affect state and federal program compliance.

### What's Changing  
The views for both digital and admin datasets will be updated to:
- Filter malformed T3 and T7 records based on the conditions above.
- Prevent AGE calculation logic from failing when date fields are null or malformed.
- Improve documentation and test coverage around these filters.

### Acceptance Criteria
- [ ] Records with `family_affiliation = 0` and `sex = 0` are excluded from all T3-related views (`tanf_t3`, `tribal_tanf_t3`, `ssp_m3`) and their admin equivalents.
- [ ] Records with `family_month = 0` and `tdrs_section_indicator` not in (1, 2) are excluded from all T7-related views (`tanf_t7`, `tribal_tanf_t7`, `ssp_m7`) and their admin equivalents.
- [ ] AGE calculation gracefully handles null or invalid date values across all affected views.
- [ ] Testing checklist has been completed and all tests pass.
- [ ] `README` is updated, if applicable.
- [ ] [Release digest](https://app.gitbook.com/o/Yd4Wv0Fi89kSKpmakWrn/s/JSIbkxujKUxhKv2cXFEg/) includes updated testing steps and relevant media (e.g., screenshots of filtered data or error-free AGE calculation logic).

### Notes  
- This ticket addresses filtering only and does not modify how files are validated during submission.
- If other malformed patterns are discovered during testing, those should be documented and triaged separately.
- Related view logic should be reviewed for similar filtering needs as part of future tech debt cleanup.

### Supporting Documentation  
- [ ] Sample malformed records from recent submissions
- [ ] SQL snippets or test queries validating filtering logic
- [ ] Screenshots demonstrating AGE field rendering correctly with null dates

### Open Questions  
- Are there any edge cases (e.g., legacy file formats) where these filters might unintentionally exclude valid data?
- Should we surface a count of filtered records in the admin interface for transparency and debugging?

### PR DESCRIPTION
**Pull request by elipe17**
## Summary of Changes
- Greatly simplified SQL view generation script
- Aggregated all view generation to single script
- Added script to automatically generate and apply views to Django DB
- Add CircleCI steps to generate and apply views dynamically
- Un-versioned views
Pull request closes #5114 

## How to Test
```
cd tdrs-frontend && docker-compose up --build
cd tdrs-backend && docker-compose up --build
```

1. Open a database viewer tool and confirm the views exist in the DB
2. Ensure that a null DOB aligns with `AGE_VALID = 0`
3. Verify the additional "where" clauses are included on the m3/t3 and m7/t7 records
4. You can also deploy this change and verify in CircleCI that the views get applied and you can verify they exist in the DB tied to the app you deployed
Screenshot 2025-06-26 at 2 31 50 PM

## Deliverables
_More details on how deliverables herein are assessed included [here](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverables)._

### [Deliverable 1: Accepted Features](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-1-Accepted-Features)

Checklist of ACs:
- [ ] Records with `family_affiliation = 0` and `sex = 0` are excluded from all T3-related views (`tanf_t3`, `tribal_tanf_t3`, `ssp_m3`) and their admin equivalents.
- [ ] Records with `family_month = 0` and `tdrs_section_indicator` not in (1, 2) are excluded from all T7-related views (`tanf_t7`, `tribal_tanf_t7`, `ssp_m7`) and their admin equivalents.
- [ ] AGE calculation gracefully handles null or invalid date values across all affected views.
- [ ] Testing checklist has been completed and all tests pass.
- [ ] `README` is updated, if applicable.
- [ ] [Release digest](https://app.gitbook.com/o/Yd4Wv0Fi89kSKpmakWrn/s/JSIbkxujKUxhKv2cXFEg/) includes updated testing steps and relevant media (e.g., screenshots of filtered data or error-free AGE calculation logic).

### [Deliverable 2: Tested Code](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-2-Tested-Code)

+ Are all areas of code introduced in this PR meaningfully tested?
  + [ ] If this PR introduces backend code changes, are they meaningfully tested?
  + [ ] If this PR introduces frontend code changes, are they meaningfully tested?
+ Are code coverage minimums met?
  + [ ] Frontend coverage: [_insert coverage %_] (see `CodeCov Report` comment in PR)
  + [ ] Backend coverage: [_insert coverage %_] (see `CodeCov Report` comment in PR)

### [Deliverable 3: Properly Styled Code](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-3-Properly-Styled-Code)

+ [ ] Are backend code style checks passing on CircleCI?
+ [ ] Are frontend code style checks passing on CircleCI?
+ [ ] Are code maintainability principles being followed?

### [Deliverable 4: Accessible](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-4-Accessibility)

+ [ ] Does this PR complete the epic? 
+ [ ] Are links included to any other gov-approved PRs associated with epic?
+ [ ] Does PR include documentation for Raft's a11y review? 
+ [ ] Did automated and manual testing with `iamjolly` and `ttran-hub` using Accessibility Insights reveal any errors introduced in this PR?

### [Deliverable 5: Deployed](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-5-Deployed)

+ [ ] Was the code successfully deployed via automated CircleCI process to development on Cloud.gov?

### [Deliverable 6: Documented](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-6-Code-documentation)

+ [ ] Does this PR provide background for why coding decisions were made?
+ [ ] If this PR introduces backend code, is that code easy to understand and sufficiently documented, both inline and overall?
+ [ ] If this PR introduces frontend code, is that code easy to understand and sufficiently documented, both inline and overall?
+ [ ] If this PR introduces dependencies, are their licenses documented?
+ [ ] Can reviewer explain and take ownership of these elements presented in this code review?

### [Deliverable 7: Secure](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-7-Secure)

+ [ ] Does the OWASP Scan pass on CircleCI?
+ [ ] Do manual code review and manual testing detect any new security issues?
+ [ ] If new issues detected, is investigation and/or remediation plan documented? 

### [Deliverable 8: User Research](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-8-User-Research)

Research product(s) clearly articulate(s):
+ [ ] the purpose of the research
+ [ ] methods used to conduct the research 
+ [ ] who participated in the research
+ [ ] what was tested and how
+ [ ] impact of research on TDP
+ [ ] (_if applicable_) final design mockups produced for TDP development

### CODE REVIEW

File: tdrs-backend/plg/grafana_views/admin_views/admin_ssp_m4.sql [hunk 1]
```diff
@@ -1,11 +1,13 @@
...
--- Generated by generate_admin_views.py
+-- Generated by generate_views.py
 
--- SQL view for admin_ssp_m4 schema
+-- SQL view for ssp_m4 schema
 -- Transformations applied:
 
-CREATE OR REPLACE VIEW "admin_ssp_m4" AS SELECT "RecordType","RPT_MONTH_YEAR","CASE_NUMBER","STRATUM","ZIP_CODE","DISPOSITION","CLOSURE_REASON","REC_SUB_HOUSING","REC_MED_ASSIST","REC_FOOD_STAMPS","REC_SUB_CC", 
+
+CREATE OR REPLACE VIEW "admin_ssp_m4" AS
```
Comment (elipe17) [inline tdrs-backend/plg/grafana_views/admin_views/admin_ssp_m4.sql]: TODO: Autogenerate and apply on the fly. No need to version sql files anymore

**Review by raftmsohani — APPROVED** [review]

**Review by jtimpe — APPROVED** [review]

**Review by jtimpe — APPROVED** [review]
Remove the generated files and should be good to go, good work!
